aws/contextTree.sh

```diff
@@ -4,688 +4,832 @@
+    for stack in "${stack_array[@]}"; do
+      # Annotate as necessary
+      if parse_stack_filename "${stack}"; then
+        modified_stack_filename="${tmp_dir}/$(fileName "${stack}")"
+        add_standard_pairs_to_stack \
+          "${stack_account:-${AWSID}}" \
+          "${stack_region}" \
+          "${stack_level}" \
+          "${stack_deployment_unit}" \
+          "${stack}" \
+          "${modified_stack_filename}"
+        modified_stack_array+=("${modified_stack_filename}")
+      else
+        modified_stack_array+=("${stack}")
       fi
-    fi
+    done
+    debug "MODIFIED_STACK_OUTPUTS=${modified_stack_array[*]}"
+    ${GENERATION_DIR}/manageJSON.sh -f "[.[].Stacks | select(.!=null) | .[].Outputs | select(.!=null) ]" -o "${COMPOSITE_STACK_OUTPUTS}" "${modified_stack_array[@]}"
+  else
+    echo "[]" > "${COMPOSITE_STACK_OUTPUTS}"
   fi
 
-  return 0
+  popTempDir
 }
 
-function checkInRootDirectory() {
-  local location="${1:-${LOCATION}}"
+function getCompositeStackOutput() {
+  local file="$1"; shift
+  local keys=("$@")
 
-  [[ ! ("root" =~ ${location}) ]] && fatalDirectory "root"
+  local patterns=()
+
+  for key in "${keys[@]}"; do
+    patterns+=(".[] | .${key}")
+  done
+  getJSONValue "${file}" "${patterns[@]}"
 }
 
-function checkInAccountDirectory() {
-  local location="${1:-${LOCATION}}"
+function getCmk() {
+  local level="$1"; shift
 
-  [[ ! ("account" =~ ${location}) ]] && fatalDirectory "account"
+  getCompositeStackOutput "${COMPOSITE_STACK_OUTPUTS}" "cmkX${level}" "cmkX${level}Xcmk"
 }
 
-function checkInProductDirectory() {
-  local location="${1:-${LOCATION}}"
+function encrypt_file() {
+  local region="$1"; shift
+  local level="$1"; shift
+  local input_file="$1"; shift
+  local output_file="$1"; shift
 
-  [[ ! ("product" =~ ${location}) ]] && fatalDirectory "product"
+  pushTempDir "${FUNCNAME[0]}_XXXXXX"
+  local tmp_dir="$(getTopTempDir)"
+  local cmk=$(getCmk "${level}")
+  local return_status
+
+  cp "${input_file}" "${tmp_dir}/encrypt_file"
+
+  (cd "${tmp_dir}"; aws --region "${region}" --output text kms encrypt \
+    --key-id "${cmk}" --query CiphertextBlob \
+    --plaintext "fileb://encrypt_file" > "${output_file}"; return_status=$?)
+
+  popTempDir
+  return ${return_status}
 }
 
-function checkInEnvironmentDirectory() {
-  local location="${1:-${LOCATION}}"
+function getBucketName() {
+  local keys=("$@")
 
-  [[ ! ("environment" =~ ${location}) ]] && fatalDirectory "environment"
+  getCompositeStackOutput "${COMPOSITE_STACK_OUTPUTS}" "${keys[@]}"
 }
 
-function checkInSegmentDirectory() {
-  local location="${1:-${LOCATION}}"
+function getBluePrintParameter() {
+  local patterns=("$@")
 
-  [[ ! ("segment" =~ ${location}) ]] && fatalDirectory "segment"
+  getJSONValue "${COMPOSITE_BLUEPRINT}" "${patterns[@]}"
 }
 
-function fatalProductOrSegmentDirectory() {
-  fatalDirectory "product or segment"
+# -- Buckets --
+
+function getOperationsBucket() {
+  getBucketName "s3XsegmentXops"
```

Comment:
This will need to account for the new bucket ids in the baseline data storage

Reply:
NOt sure if I actually changed this - maybe just missed with the baseline changes?